de.dlr.sc.overtarget.language.ui/src/de/dlr/sc/overtarget/language/ui/handler/TargetActivationHandler.java

```diff
@@ -64,13 +78,32 @@ public Object execute(ExecutionEvent event) throws ExecutionException {
 			String outputConfig = getOutputConfigurations(project);
 			IFile targetFile;
 			String outputPath = outputConfig.replace(".", "");
+			
+			URI uri = URI.createPlatformResourceURI(file.getFullPath().toString(), true);
+			ResourceSet rs = resourceSetProvider.get(project);
+			Resource r = rs.getResource(uri, true);
+			
+			//to check if the tmodel is a target library
+			TargetFile target = (TargetFile) r.getContents().get(0);
```

Comment:
Couldn’t you have just checked what kind of object the target is? Something like „target instanceof TargetLibarary“? I‘m just wondering... Out of interest

Reply:
Yes, I wasn't that smart, this works much easier 👍